core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -296,4 +297,72 @@ public final int hashCode() {
      * HashCode for the subclass of {@link SuggestionBuilder} to implement.
      */
     protected abstract int doHashCode();
-}
\ No newline at end of file
+
+
+    /** An enum representing the valid suggest modes. */
+    public enum SuggestMode implements NamedEnum {
```

Comment:
Its great to have these as enums, especially with all the added javadoc. If they are specific to the TermsSuggestionBuilder (haven't checked), I would move them there though.